microprofile/server/src/main/java/io/helidon/microprofile/server/Server.java

```diff
@@ -315,7 +320,7 @@ private void resourcesFromContainer() {
         private SeContainer createContainer(ClassLoader classLoader) {
             // not in CDI
             Weld initializer = new Weld();
-            initializer.addBeanDefiningAnnotations(Path.class);
+            initializer.addBeanDefiningAnnotations(Path.class, Provider.class);
```

Comment:
As you said in chat, we want to load only providers that are annotated using a bean defining annotation. This change modifies that behavior by making all of the having a bean defining annotation.

Reply:
I just wanted to make sure it works exactly the way resources work. It is my understanding that this code ensures that any resources (and now providers) that have been explicitly registered via Builder are also registered with the CDI container. 

I think that makes sense, and is separate from CDI discovery, which is done by the extension and expects providers to be proper managed beans.

Reply:
The problem is that this is a backward incompatible change and may have unexpected behavior.
If you add `@ApplicationScoped` to a class annotated as `@Provider`, it will be automatically picked up by your changes.
Adding `Provider` as a bean defining annotation is a step too far, as any provider in the code base would get added automatically, and that may not be desired. I would keep the original agreed approach that only beans are added automatically

Reply:
Perhaps this is something we can do in 2.0 where the scanning code is already incompatible

Reply:
But that's the thing -- it doesn't, just like the resources annotated with`@Path` do not get automatically picked **unless** they are also annotated with `@ApplicationScoped` or `@RequestScoped`.

The only thing the code above does is **allow** you to register classes that do not have any standard bean defining annotations with the bean manager, which happens a little bit later in the code:

```
            // add resource classes explicitly configured without CDI annotations
            this.resourceClasses.stream()
                    .filter(this::notACdiBean)
                    .forEach(initializer::addBeanClasses);

            // add provider classes explicitly configured without CDI annotations
            this.providerClasses.stream()
                    .filter(this::notACdiBean)
                    .forEach(initializer::addBeanClasses);
```
I've tried running my demo app using this implementation, and my filter is only discovered and registered automatically if I add `@ApplicationScoped` annotation as well. None of the other `@Provider` classes get picked up automatically, even though I depend on MP bundle and am bringing in a ton of dependencies.

The bottom line is that because Jersey treats resources and providers the same (they are all returned from the same `Application.getClasses` method), we should treat them the same as well and allow for one what we already allow for the other (in this case, registration of explicitly added resources and providers that are **not** proper managed beans with bean manager). I thought about not even separating them out and using a single list to collect all resource and provider classes, but decided against it as we have a few places where we log warning if there are no resources defined for the app, so having them separate is helpful.

Reply:
I just realized this is the 1.4 branch, where we handle resources differently from 2.0. On `master` some of these assumptions are no longer valid. Sorry about that.
We probably can keep the same function for 2.0 when it comes to `@Provider`, though it does work differently for `@Path`